routing/matcher_test.go

```diff
@@ -1,79 +1,1161 @@
+	testMatch(t, "GET", "/sls/herren/nike", "https://herren-sls.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/damen/nike", "https://damen-sls.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/men/nike", "https://herren-en.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/women/nike", "https://damen-en.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/men/nike", "https://herren-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/women/nike", "https://damen-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/men/nike", "https://herren-en.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/women/nike", "https://damen-en.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/men/nike", "https://herren-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/women/nike", "https://damen-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/herren/nike/sports", "https://herren.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/damen/nike/sports", "https://damen.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/herren/nike/sports", "https://herren-async.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/damen/nike/sports", "https://damen-async.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/herren/nike/sports", "https://herren-sc.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/damen/nike/sports", "https://damen-sc.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/herren/nike/sports", "https://herren-sls.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/damen/nike/sports", "https://damen-sls.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/men/nike/sports", "https://herren-en.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/women/nike/sports", "https://damen-en.layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/men/nike/sports", "https://herren-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls-async/women/nike/sports", "https://damen-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/men/nike/sports", "https://herren-en.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sc/women/nike/sports", "https://damen-en.compositor-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/men/nike/sports", "https://herren-en.streaming-layout-service.my-department.example.org")
+	testMatch(t, "GET", "/sls/women/nike/sports", "https://damen-en.streaming-layout-service.my-department.example.org")
```

Comment:
it was generated with vim, except for the post and pdp requests :)
actually, we don't need all these regardless of the generating method. it was useful during working on the route lookup itself